### ISSUE TITLE
Extension indexes the whole drive when no folder is opened

### ISSUE THREAD

**Issue by WorldofBay** [microsoft/vscode-cpptools#2216]
I'm on windows but I'm sure it is patform-independent, maybe someone wants to confirm.

Expected CPU and Disk load on empty VS Code: idle
Experienced CPU load on empty VS Code with C/C++ extension: up to 1 full core
Experienced Disk load on my SSD: spikes up to 30%

Opening a c++ file in VS Code triggers the C/C++ extension to parse files even without a folder opened. The problem is that parsing files does not stop when there is no c++ file left open.

That leads to C/C++ extension using an entire CPU core on an empty VS Code (no folder, no files).

The question on that is what does the extension even parse? Does it parse the whole machine?
The highest number of files i saw when clicking on the parsing icon was >4 million.

How to reproduce:
- open empty VS Code with C/C++ extension installed and enabled
- create new anonymous file using ctrl+N (or use the default open anonymous file "Untitled-1")
- change file language to "c++"
- close file
- watch CPU load shoot up to 1 core + slightly more

**Comment by bobbrow** [comment]
Can you set the `"C_Cpp.loggingLevel"` setting to `"Debug"`?  Then open an empty VS Code and a cpp file and watch the Output window for messages that say what we are parsing.

On Windows, I would only expect us to parse the Visual Studio/Windows SDK headers on your machine that are included by the source file you opened.  We will also index all of those headers, but shouldn't parse them unless you include them in your source file.  But if you happen to have any of the `"C_Cpp.default.*` settings set globally, we might do more work than that.

**Comment by sean-mcmanus** [comment]
This might be partially due to bug https://github.com/Microsoft/vscode-cpptools/issues/2156  (causing system source files to be parsed) , which should be fixed in the pending 0.17.6 (due today, unless there's a delay).

**Comment by WorldofBay** [comment]
I have not set any of the C_Cpp.default options yet.

**Output:**

```
initialized
workspace/didChangeConfiguration
IntelliSense Engine = Default.
The extension will use the Tag Parser for IntelliSense when #includes don't resolve.
Autocomplete is enabled.
Error squiggles are enabled.
File exclude: **/.git
File exclude: **/.svn
File exclude: **/.hg
File exclude: **/CVS
File exclude: **/.DS_Store
File exclude: **/*.git
File exclude: **/*.svn
File exclude: **/*.hg
File exclude: **/*.DS_Store
File exclude: **/*.vcxproj
File exclude: **/.vscode
Search exclude: **/node_modules
Search exclude: **/bower_components
Search exclude: **/*.git
Search exclude: **/*.svn
Search exclude: **/*.hg
Search exclude: **/CVS
Search exclude: **/*.DS_Store
Search exclude: **/*.vcxproj
Search exclude: */_build.70/**/*.*
Search exclude: */_build.50/**/*.*
Search exclude: **/*log
Search exclude: **/.vscode
cpptools/queryCompilerDefaults
cpptools/activeDocumentChange
cpptools/textEditorSelectionChange
cpptools/didChangeFolderSettings
Code browsing service initialized
  Folder: C:/ will be indexed
Discovering files...
  Processing folder (recursive): C:/
Database safe to open
Database safe to open
Database safe to open
Database safe to open
  Discovering files: 1064495 file(s) processed
  0 file(s) removed from database
```
and then a long list of
`Removed file`
of temporary files in C:\ProgrammData, C:\Programm Files and C:\Users
```
Done discovering files.
Parsing remaining files...
  Parsing: 0 files(s) processed
Done parsing remaining files.
```
Then CPU spike is done (~5 minutes).

So it actually didn't parse anything this time. It did previously (according to the gui icon).
I also noticed that the CPU spike was done a lot faster this time. 
Might be that the update in between my posts improved it already a lot.

The file i used to trigger C/C++ extension is the default open "Untitled-1" file with no content changed to language c++ (no further action).

**Comment by bobbrow** [comment]
It should be faster the second time because we only re-parse files when the timestamp changes.  But this is very bad.  I just opened an empty window and saw the same thing.  The extension is trying to index the whole drive.  The CPU spike the second time is from the extension indexing the drive.  We'll get this fixed for the next release.

**Comment by sean-mcmanus** [comment]
@bobbrow On Windows, it will also cause the main process to crash from an out of memory if there are too many files on the drive. The bug doesn't appear to repro on Linux, but if it did, it would eat up most of the memory on the machine due to the 64-bitness.

**Comment by sean-mcmanus** [comment]
The indexing of the root drive without a workspace open has been fixed with 0.17.7.
